ansible-collection-microsoft-sql/ansible-collection-microsoft-sql.spec

```diff
@@ -308,71 +140,54 @@ for rolename in %{rolenames}; do
         fi
     fi
 done
-rm %{rolename5}/tests/modules
-rm %{rolename5}/tests/module_utils
-rm %{rolename5}/tests/playbooks/roles
 
 # transform ambiguous #!/usr/bin/env python shebangs to python3 to stop brp-mangle-shebangs complaining
 find -type f -executable -name '*.py' -exec \
      sed -i -r -e '1s@^(#! */usr/bin/env python)(\s|$)@#\13\2@' '{}' +
 
 %build
+# Convert README.md to README.html in the source roles
 readmes=""
-for role in %{rolenames}; do
-    readmes="${readmes} $role/README.md"
+for rolename in %{rolenames}; do
+    readmes="${readmes} $rolename/README.md"
 done
 sh md2html.sh $readmes
 
 mkdir .collections
-%if 0%{?rhel}
-# Convert the upstream collection readme to the downstream one
-%{SOURCE998} lsr_role2collection/collection_readme.md
-%endif
-./galaxy_transform.py "%{collection_namespace}" "%{collection_name}" "%{collection_version}" "Red Hat Enterprise Linux System Roles Ansible Collection" > galaxy.yml.tmp
+# Copy README.md for the collection build
+cp mssql/.collection/README.md lsr_role2collection/collection_readme.md
+# Copy galaxy.yml for the collection build
+cp mssql/.collection/galaxy.yml ./
+# Ensure the correct entries in galaxy.yml
+./galaxy_transform.py "%{collection_namespace}" "%{collection_name}" "%{collection_version}" "The Ansible role that manages Microsoft SQL Server" > galaxy.yml.tmp
 mv galaxy.yml.tmp galaxy.yml
 
-for role in %{rolenames}; do
-    python3 lsr_role2collection.py --role "$role" --src-path "$role" \
-        --src-owner %{name} --subrole-prefix %{subrole_prefix} --dest-path .collections \
-        --readme lsr_role2collection/collection_readme.md \
-        --namespace %{collection_namespace} --collection %{collection_name}
-done
-
-# copy requirements.txt and bindep.txt from auto-maintenance/lsr_role2collection
-if [ -f lsr_role2collection/collection_requirements.txt ]; then
-    cp lsr_role2collection/collection_requirements.txt \
-       .collections/ansible_collections/%{collection_namespace}/%{collection_name}/requirements.txt
-fi
-if [ -f lsr_role2collection/collection_bindep.txt ]; then
-    cp lsr_role2collection/collection_bindep.txt \
-       .collections/ansible_collections/%{collection_namespace}/%{collection_name}/bindep.txt
-fi
+# Declare the array containing collection rolenames to convert roles to
+declare -A COLLECTION_ROLENAMES=(%{collection_rolenames})
 
-rm -f .collections/ansible_collections/%{collection_namespace}/%{collection_name}/tests/sanity/ignore-2.9.txt
```

Comment:
I'm curious about the future of mssql family which is going to use this spec file. Since you keep the array of rolenames, I assume the family member size would increase. If the assumption is valid, it might be a good idea to keep the framework for `requirements.txt`, `bindep.txt`, and `ignore-#.#.txt` although the current mssql needs none of them.

Reply:
I want to keep the spec file as minimalistic as possible for now, now this framework is not needed so I think there is no need to keep it, it would only cause confusion. I keep in mind that this exists in the linux-system-roles spec file and will copy it to here if needed in future.